### DISCUSSION TITLE
Flag to request removal of a package from a grid

### DISCUSSION THREAD

**Discussion by ShreehariVaasishta** [djangopackages/djangopackages#763] [Feature Ideas]
**Is your feature request related to a problem? Please describe.**
This discussion is related to the issue https://github.com/djangopackages/djangopackages/issues/96. Hence i would like to discuss this here before moving forward with the development.

**Describe the solution you'd like**
Any authenticated user on the platform should be able to flag a package. A user can flag a particular package only once.
If a package is flagged with more than `n` (**n** can be decided) number of users, then it can be excluded from the list.

**Solution in detail**
1. Create a model named `FlaggedPackages ` with **package** and **user** as ForeignKeys. It will have a unique constraint on package and user. This helps to check whether a user has already flagged a package or not.
2. If a package has been flagged more than **n** number of times, we can automatically exclude them from the filter() query wherever it is being used. **OR** We can have a field named **active** `(default=True)` on the Package model which will be set to False when there are **n** of flags.

**Comment by jayden-clark** [comment]
Just following up on this issue.

The solution that Shreehari gives is a good one, although rather than having a package be automatically removed from the filter() query after n number of times, it could be worth setting the field named **active** to `(default=False)` and then adding a method to the `FlaggedPackages` model named `approve` to allow for further verification from an author role. 
Perhaps add a field named `reason` to users to add their reason for flagging to be viewed by an author.

Once the flag is approved by an author, a flag will be displayed next to the package.

A solution like this would prevent unwanted flagging of packages that aren't yet but soon to be compatible with newer Python versions for example.
